packages/data/addon/adapters/fact-interface.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+
+export type RequestV1 = TODO;
+
+export type RequestOptions = {
+  clientId?: string;
+  customHeaders?: Dict<string>;
+  timeout?: number;
+  page?: number;
+  perPage?: number;
+  format?: string;
+  cache?: boolean;
+  queryParams?: Dict<string>;
+  dataSourceName?: string;
+};
+
+export const SORT_DIRECTIONS = ['desc', 'asc'];
+
+export type Parameters = Dict<string>;
+
+export type SortDirection = typeof SORT_DIRECTIONS[number];
+
+export type ColumnType = 'metric' | 'dimension' | 'timeDimension';
```

Comment:
should this be `time-dimension`?

Reply:
This was moved over as is from the v2 adapter. The v2 RFC has `timeDimension`. This is also how elide formats it in the metadata endpoints. Do you know where we are using the dasherized form?

Reply:
https://github.com/yahoo/navi/blob/9b9f0a31c52817ee88ced057d71f89b7c8d8c628/packages/data/addon/services/bard-metadata.js#L19

Pushing/pulling from our metadata store

Reply:
Also in
https://github.com/yahoo/navi/blob/request-v2/packages/core/addon/models/bard-request-v2/fragments/base.js#L19

Reply:
I am going to make an issue for this

Reply:
https://github.com/yahoo/navi/issues/852